src/main/scala/beam/utils/protocolvis/Extractors.scala

```diff
@@ -0,0 +1,36 @@
+package beam.utils.protocolvis
+
+import beam.utils.protocolvis.MessageReader.RowData
+
+/**
+  * @author Dmitry Openkov
+  */
+object Extractors {
+
+  def byPerson(personId: String): Function[Iterator[RowData], IndexedSeq[RowData]] = { stream =>
+    val (_, seq) = stream.foldLeft(Set.empty[Long] -> IndexedSeq.empty[RowData]) {
+      case ((ids, seq), row) =>
+        if (isSenderOrReceiver(personId, row)) (if (row.triggerId >= 0) ids + row.triggerId else ids, seq :+ row)
+        else if (ids.contains(row.triggerId)) (ids + row.triggerId, seq :+ row)
```

Comment:
No, because triggerId is pushed down to the message sequence and the initial person may not be involved directly in some interactions within the same sequence of messages (with the same triggerId).